manual/manual/html_processing/src/process_api.ml

```diff
@@ -0,0 +1,414 @@
+(* Post-processing the HTML of the OCaml API.
+
+   San Vu Ngoc, 2020
+
+   requires Lambdasoup
+
+   cd ..; make web
+
+ *)
+
+open Soup
+open Printf
+open Common
+
+let compiler_libref = ref false
+(* set this to true to process compilerlibref instead of libref *)
+
+let libref = ref ""
+let dst_dir = ref ""
+let src_dir = ref ""
+
+let set_dirs () =
+  libref := if !compiler_libref then "compilerlibref" else "libref";
+  dst_dir :=
+    if !compiler_libref then !libref |> with_dir api_dir
+    else api_dir;
+  src_dir := !libref |> with_dir html_maindir
+
+let () = set_dirs ()
+
+let favicon () = if !compiler_libref then "../favicon.ico" else "favicon.ico"
+
+(* HTML code for the search widget. We don't add the "onchange" event because it
+   forces to click twice to an external link after entering text. *)
+let search_widget with_description =
+  sprintf "<div class=\"api_search\"><input type=\"text\" name=\"apisearch\" id=\"api_search\"
+	 oninput    = \"mySearch(%b);\"
+         onkeypress = \"this.oninput();\"
+         onclick    = \"this.oninput();\"
+	 onpaste    = \"this.oninput();\">
+<img src=\"search_icon.svg\" alt=\"Search\" class=\"svg\" onclick=\"mySearch(%b)\">%s</div>
+<div id=\"search_results\"></div>" with_description with_description
```

Comment:
If you use `{|<div class="api_search"...|}` then you want to quote all the double-quotes.

Reply:
ok, done